main/activity-log.js

```diff
@@ -1,47 +1,46 @@
 'use strict'
 
-/** @typedef {import('./typings').ActivityEvent}  ActivityEvent */
-/** @typedef {import('./typings').ActivityEntry}  ActivityEntry */
+/** @typedef {import('./typings').Activity} Activity */
+/** @typedef {import('./typings').RecordActivityOptions}  RecordActivityOptions */
 
 const Store = require('electron-store')
+const crypto = require('node:crypto')
+
 const activityLogStore = new Store({
   name: 'activity-log'
 })
 
 class ActivityLog {
   #entries
-  #lastId
 
   constructor () {
     this.#entries = loadStoredEntries()
-    this.#lastId = Number(this.#entries.at(-1)?.id ?? 0)
   }
 
   /**
-   * @param {ActivityEvent} args
-   * @returns {ActivityEntry}
+   * @param {RecordActivityOptions} args
+   * @returns {Activity}
    */
-  recordEvent ({ source, type, message }) {
-    const nextId = ++this.#lastId
-    /** @type {ActivityEntry} */
-    const entry = {
-      id: String(nextId),
+  recordActivity ({ source, type, message }) {
```

Comment:
Since this is a method of `ActivityLog` class, maybe we can shorten the name to simply `record`?

```suggestion
  record ({ source, type, message }) {
```

Reply:
good idea!